website/docs/gator.md

```diff
@@ -0,0 +1,180 @@
+---
+id: gator
+title: The gator CLI
+---
+
+The `gator` CLI is a tool for testing Gatekeeper ConstraintTemplates and
+Constraints on Kubernetes objects.
+
+We plan on adding more subcommands in the future.
+
+## Installation
+
+To install `gator`, you may either
+[download the binary](https://github.com/open-policy-agent/gatekeeper/releases/tag/v3.7.0)
+relevant to your system or build it directly from source.
+
+To build from source:
+```
+go get github.com/open-policy-agent/gatekeeper/cmd/gator
+```
+
+## The `gator test` subcommand
+
+### Writing Test Suites
+
+`gator test` organizes tests into three levels: Suites, Tests, and Cases:
+
+- A Suite is a file which defines Tests.
+- A Test declares a ConstraintTemplate, a Constraint, and Cases to test the
+  Constraint.
+- A Case defines an object to validate and whether the object is expected to
+  pass validation.
+
+Any file paths declared in a Suite are assumed to be relative to the Suite
+itself. Absolute paths are not allowed. Thus, it is possible to move around a
+directory containing a Suite, and the files it uses for tests.
+
+### Suites
+
+[An example Suite file](https://github.com/open-policy-agent/gatekeeper-library/blob/8765ec11c12a523688ed77485c7a458df84266d6/library/general/allowedrepos/suite.yaml).
+
+To be valid, a Suite file must declare:
+```yaml
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+```
+
+This differentiates Suites from "real" Kubernetes objects. `gator test` silently
```

Comment:
I'm not sure what you mean by differentiating here.  What about it doesn't match a real k8s object?

Reply:
You can't/shouldn't `kubectl apply` a Suite.

Reply:
> You can't/shouldn't kubectl apply a Suite

I get the can't `kubectl apply` part, but mostly because the CRD doesn't exist on a cluster.  Further, k8s would just do nothing with the resource since there are no meaningful controllers watching it.

> It also allows the CLI to identify suites, as it means they can't possibly be real Kubernetes objects.

Can you explain this?  The `Suite` kind seems to be KRM, so how would we know it's not a "real" k8s type?  Couldn't it just be a CR?

Reply:
Removed sentence.